### ISSUE TITLE
Get screenshots of lots of sites when they go down

### ISSUE THREAD

**Issue by lippytak** [codeforamerica/snap-it-up#11]
It will make for a more compelling essay.

**Comment by daguar** [comment]
@Mr0grog mentioned this a while back right? Maybe he knows how to do it easily...

**Comment by Mr0grog** [comment]
Oh yeah, that was here: https://github.com/codeforamerica/project-ideas/issues/43#issuecomment-65971544  I think that service is still running…

But at the time, my thinking was: what’s the actual value there? But I guess it’s illustrative if you’re writing about the issues with these services going down. Just not really as part of a dashboard. Is that something this app should do? It _should_ be easy enough for someone to write a separate script that calls that service when a SNAP site goes down. (Well, excepting the sites that require automated browsing, which would be a bit more complicated.)

**Comment by lippytak** [comment]
![screen shot 2015-01-30 at 10 47 07 am](https://cloud.githubusercontent.com/assets/2533112/5981265/859ba542-a86d-11e4-91fe-e227551f9fb9.png)

mybenefitscalwin.org

**Comment by lippytak** [comment]
![screen shot 2015-02-02 at 9 30 40 pm](https://cloud.githubusercontent.com/assets/2533112/6015042/4b6bba9c-ab24-11e4-8f96-297cd73b5847.png)

Indiana

**Comment by lippytak** [comment]
![screen shot 2015-02-03 at 7 33 17 pm](https://cloud.githubusercontent.com/assets/2533112/6034790/f3a054ea-abdb-11e4-96cd-977f46e212c3.png)

![screen shot 2015-02-03 at 7 33 24 pm](https://cloud.githubusercontent.com/assets/2533112/6034791/f3a36a90-abdb-11e4-8045-351ce1cffed2.png)

Illinois

**Comment by lippytak** [comment]
![screen shot 2015-02-03 at 7 39 26 pm](https://cloud.githubusercontent.com/assets/2533112/6034813/84107b0e-abdc-11e4-81ea-4b5b44d4b33f.png)

Virginia https://commonhelp.virginia.gov/access/

**Comment by lippytak** [comment]
![arizona](https://cloud.githubusercontent.com/assets/2533112/6142386/51999dca-b16a-11e4-9dff-132f9b2e2ec3.png)
Arizona

**Comment by lippytak** [comment]
![screen shot 2015-02-10 at 9 19 17 pm](https://cloud.githubusercontent.com/assets/2533112/6142394/7f54e1ca-b16a-11e4-80ad-9a4524f769af.png)
New Hampshire

**Comment by alanjosephwilliams** [comment]
Did you get one of Indiana yesterday??

On Tue, Feb 10, 2015 at 9:19 PM, Jake Solomon notifications@github.com
wrote:

> [image: screen shot 2015-02-10 at 9 19 17 pm]
> https://cloud.githubusercontent.com/assets/2533112/6142394/7f54e1ca-b16a-11e4-80ad-9a4524f769af.png
> New Hampshire
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/codeforamerica/snap-it-up/issues/11#issuecomment-73835536
> .

## 

t:  @alanjosephwilli
p: 817 713 6264

**Comment by lippytak** [comment]
@alanjosephwilliams already here

**Comment by Mr0grog** [comment]
I added webhook code to do this automatically and save all screenshots to S3 in our ‘snap-snapshots’ bucket. However, it seems like you only get one alert group per monitor in @pingometer. Also, it seems like you have to set up each monitor individually (oof). I’ve e-mailed their support about those issues.

**Comment by pingometer** [comment]
Very cool!

You can add as many groups as your want to each monitor - there's no limit
there. I replied to your ticket (re: bulk-adding a new group to all
monitors).

**Comment by Mr0grog** [comment]
Thanks, @pingometer. Looks like this is actually working now:

![](https://s3-us-west-2.amazonaws.com/snap-snapshots/CA-54bc8944be653d3f86065dc5-2015-02-16T22%3A41%3A20%2B00%3A00)
https://s3-us-west-2.amazonaws.com/snap-snapshots/CA-54bc8944be653d3f86065dc5-2015-02-16T22%3A41%3A20%2B00%3A00

^ that was autogenerated from MyBenefitsCalwin going up and down this afternoon.

I wonder if we’ll always have trouble snapshotting that one since we are using Phantom… may need to look into other ways of snapshotting; maybe via saucelabs or something. :(

**Comment by Mr0grog** [comment]
(Also, you can list all the snapshots in XML by going to https://s3-us-west-2.amazonaws.com/snap-snapshots/)

**Comment by daguar** [comment]
Can Selenium with a Firefox driver be used to address that?

**Comment by Mr0grog** [comment]
Maybe; it’s just approximately five times more complicated. Don’t know how well it’ll work on Heroku, either.

**Comment by pingometer** [comment]
Can you "fake" the user agent string? If you make it appear like it's
coming from Firefox or IE, it should let you through.

**Comment by Mr0grog** [comment]
@pingometer Ah! Good point, I’ll have to give that a try when it’s back up. Pretty sure you can customize the UA string in Phantom.

**Comment by pingometer** [comment]
@Mr0grog Yup, looks like it's part of `page.settings`:
http://phantomjs.org/api/webpage/property/settings.html

**Comment by lippytak** [comment]
![screen shot 2015-02-18 at 5 52 15 pm](https://cloud.githubusercontent.com/assets/2533112/6260649/e5077230-b796-11e4-9065-a2c524b14df6.png)

![screen shot 2015-02-18 at 5 52 00 pm](https://cloud.githubusercontent.com/assets/2533112/6260646/debb2390-b796-11e4-872d-45655be310a9.png)

Reason to go deeper ^

**Comment by Mr0grog** [comment]
Also, as a diagnostic tool, I’ve got heroku scheduler now set up to run a script that will snapshot all the sites in the middle of the night. A lot of them are down during that time, so it’ll give us a nice way to get a quick look at what they all look like (and an easy thing to check to see if we’re mis-monitoring them).

**Comment by lippytak** [comment]
![screen shot 2015-02-26 at 11 06 01 am](https://cloud.githubusercontent.com/assets/2533112/6399471/7ac6904c-bda7-11e4-935f-e3709bbe592e.png)

Louisiana has been down for a month http://dcfs.la.gov/index.cfm?md=pagebuilder&tmp=home&pid=407

**Comment by Mr0grog** [comment]
…I don’t think it’s _ever_ been up for the entire lifetime of this project, has it?

**Comment by lippytak** [comment]
Correct. Just realized I never captured it here. But looks like it will be
back up monday according to the message! (The endless monday)

**Comment by lippytak** [comment]
![screen shot 2015-03-03 at 10 07 59 am](https://cloud.githubusercontent.com/assets/2533112/6468898/526f8452-c18d-11e4-913c-e2b885456804.png)

https://www.c4yourself.com/c4yourself/index.jsp

I know we're getting these automatically, but I'm on my comp and I just liked this one so why not.

**Comment by Mr0grog** [comment]
No, that’s good—seems like we are having some trouble screenshotting that site. Should probably see what’s going in in PageSnap… https://snap-snapshots.s3-us-west-2.amazonaws.com/CA-54a79f58be653d4687bf5eef-UP-54f3512419ad89639917e538.png

**Comment by Mr0grog** [comment]
Well, now I know what’s wrong with PageSnap, but no easy fix. Seems like there’s an SSL handshake error on C4yourself and some bug in Phantom won’t let you get around it: https://github.com/ariya/phantomjs/issues/12181

**Comment by Mr0grog** [comment]
Now generating screenshots through SlimerJS/Firefox (which doesn’t have the SSH bug Phantom does) on an extraordinarily hacked together EC2 instance (couldn’t get XVFB and Firefox running on Heroku). Same API as pagesnap, just running Firefox. Don’t really want to maintain that or make it super robust if I don’t have to. Probably better to be using a service like Browserstack or Saucelabs. Anybody know if there’s any chance we could get some non-profit help/credit from either of them? :)
